jaeger-core/src/main/java/io/jaegertracing/JaegerSpan.java

```diff
@@ -257,6 +297,15 @@ public JaegerSpan log(String event) {
   @Override
   public JaegerSpan log(long timestampMicroseconds, String event) {
     synchronized (this) {
+      if (finished) {
+        if (tracer.isLoggingStackTraces()) {
+          log.warn("Span has already been finished; cannot log.", new RuntimeException("Called from:"));
+        } else {
+          log.warn("Span has already been finished; cannot log.");
+        }
+        return this;
+      }
```

Comment:
I suggest to DRY it by moving this to a method `spanAlreadyFinished(String funcName)` and then replacing the checks with 
```
if (spanAlreadyFinished("setTag")) {
    return this;
}
```